src/index/index.c

```diff
@@ -30,7 +30,7 @@ int compareStrings(SIValue *a, SIValue *b) {
 }
 
 int compareNumerics(SIValue *a, SIValue *b) {
-  double diff = a->doubleval - b->doubleval;
+  double diff = SIValue_Compare(*a, *b);
```

Comment:
This is likely to cause some performance degradation, so I'd like to improve the logic here soon.